packages/nep141-erc20/src/bridged-nep141/sendToEthereum/index.js

```diff
@@ -85,6 +86,93 @@ export function checkStatus (transfer) {
   }
 }
 
+// Recover transfer from a withdraw tx hash
+// Track a new transfer at the completedStep = WITHDRAW so that it can be unlocked
+export async function recover (withdrawTxHash, sender='todo') {
+  const decodedTxHash = utils.serialize.base_decode(withdrawTxHash)
+  const nearAccount = await getNearAccount()
+  const withdrawTx = await nearAccount.connection.provider.txStatus(
+    // TODO: when multiple shards, the sender should be known in order to query txStatus
+    decodedTxHash, sender
+  )
+  sender = withdrawTx.transaction.signer_id
+
+  if (withdrawTx.status.Unknown) {
+    // Transaction or receipt not processed yet
+    throw new Error('Withdraw transaction pending: %s', withdrawTxHash)
+  }
+
+  // Check status of tx broadcasted by wallet
+  if (withdrawTx.status.Failure) {
+    throw new Error('Withdraw transaction failed: %s', withdrawTxHash)
+  }
+
+  const receiptIds = withdrawTx.transaction_outcome.outcome.receipt_ids
+
+  if (receiptIds.length !== 1) {
+    throw new Error(
+      `Withdrawal expects only one receipt, got ${receiptIds.length}.
+      Full withdrawal transaction: ${JSON.stringify(withdrawTx)}`
+    )
+  }
+
+  const txReceiptId = receiptIds[0]
+
+  const successReceiptId = withdrawTx.receipts_outcome
+    .find(r => r.id === txReceiptId).outcome.status.SuccessReceiptId
+  const txReceiptBlockHash = withdrawTx.receipts_outcome
+    .find(r => r.id === successReceiptId).block_hash
+
+  const receiptBlock = await nearAccount.connection.provider.block({
+    blockId: txReceiptBlockHash
+  })
```

Comment:
This appears to be largely copy-pasted from `withdraw`. How complicated would it be to extract to extract to a function that both functions call?

Reply:
Agree, got a todo to look into it. The new withdraw tx flow lives in https://github.com/near/rainbow-bridge-client/pull/7 so it can be done after merge and rebase.